sunpy/net/base_client.py

```diff
@@ -31,9 +109,56 @@ def __init_subclass__(cls, *args, **kwargs):
         This means that Fido can use the clients internally.
         """
         super().__init_subclass__(**kwargs)
+
         # We do not want to register GenericClient since its a dummy client.
-        if cls.__name__ != 'GenericClient':
-            cls._registry[cls] = cls._can_handle_query
+        if cls.__name__ in ('GenericClient'):
+            return
+
+        cls._registry[cls] = cls._can_handle_query
+
+        if hasattr(cls, "_attrs_module"):
+            from sunpy.net import attrs
+
+            name, module = cls._attrs_module()
+            module_obj = importlib.import_module(module)
+
+            existing_mod = getattr(attrs, name, None)
+            if existing_mod and existing_mod is not module_obj:
+                raise NameError(f"{name} has already been registered as an attrs name.")
+
+            setattr(attrs, name, module_obj)
+
+            if name not in attrs.__all__:
+                attrs.__all__.append(name)
+
+        # Register client attrs after it has regsitered its own attrs
+        from sunpy.net import attr
+        attr.Attr.update_values(cls.register_values())
+
+    def __repr__(self):
+        """
+        Returns the normal repr plus the pretty client __str__.
+        """
+        return object.__repr__(self) + "\n" + str(self)
```

Comment:
Do we want to print the whole table on repr?

Reply:
Half the table? A slice of the table?